zproject/backends.py

```diff
@@ -2629,6 +2693,12 @@ def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
 
             # We want the IdP name to be accessible from the social pipeline.
             self.strategy.session_set("saml_idp_name", idp_name)
+            session_index = saml_document.get_session_index()
+            if session_index is None:
+                self.logger.info(
+                    "/complete/saml/: IdP did not provide SessionIndex in the SAMLResponse."
+                )
```

Comment:
Can you add a comment for this block documenting what this means?

My current understanding is that this is normal for an IdP that does not intend to do SP-initiated SLO; is that correct? If so, I wonder if we want to check if that setting is enabled on this Zulip server, and make it an error that aborts the request if so, and a debug-level log if not? 

That seems like it might be more conducive to helping self-hosters verify that they've setup SP-initiated SLO correctly.

Reply:
> My current understanding is that this is normal for an IdP that does not intend to do SP-initiated SLO; is that correct? 

Hmm, not quite, they actually just generally always send `SessionIndex` regardless of settings i believe. Tested with Auth0 right now since i have it set up, and it sends by default, Keycloak I remember also just always sends it and I didn't verify but im pretty sure Okta was also sending it when i was fiddling with it. It's not strictly needed though, so perhaps there are providers that neglect to send it - that's why i left that `if` there, as a "hypothetically possible, but unusual, so `logger.info` about it". 

`SessionIndex` is for generally having a clear session identifier, not only with SLO-specific intent. From the SAML spec (https://docs.oasis-open.org/security/saml/v2.0/saml-core-2.0-os.pdf):
![image](https://github.com/zulip/zulip/assets/45007152/8b7ed456-0d14-4a0e-88bb-57b5c3e6632f)

I added a comment like this now:
```
            if session_index is None:
                # In general IdPs will always provide a SessionIndex, but we can't know
                # if some providers might not send it, so we allow it but log the event.
                self.logger.info(
                    "/complete/saml/: IdP did not provide SessionIndex in the SAMLResponse."
                )
```

Reply:
Sure, this works for me.